exercises/practice/pig-latin/test_pig_latin.c

```diff
@@ -2,210 +2,192 @@
 #include "pig_latin.h"
 #include <stdlib.h>
 
+static char *res = NULL;
 void setUp(void)
 {
 }
 
 void tearDown(void)
 {
+   if (res)
+      free(res);
+   res = NULL;
```

Comment:
This seems like a bit of a tautology here, in that `res` is set to `NULL` even if it already is.
Would it make more sense, semantically, to set `res` to `NULL` in the `setup()` ?

Reply:
I based it on phone number. And I think it makes sense to do it here as part of a thorough cleanup. See [here](https://stackoverflow.com/questions/1025589/setting-variable-to-null-after-free#:~:text=free()%20is%20a%20library,will%20be%20set%20to%20NULL.)